p2p.go

```diff
@@ -11,105 +11,105 @@ import (
-	PrivateNetworkPSK:        "",
-}
+// ProtocolDHT is the IoTeX DHT protocol name
+const ProtocolDHT protocol.ID = "/iotex"
+
+type (
+	// HandleBroadcast defines the callback function triggered when a broadcast message reaches a host
+	HandleBroadcast func(ctx context.Context, data []byte) error
+
+	// HandleUnicast defines the callback function triggered when a unicast message reaches a host
+	HandleUnicast func(ctx context.Context, w io.Writer, data []byte) error
+
+	// Config enumerates the configs required by a host
+	Config struct {
+		HostName                 string          `yaml:"hostName"`
+		Port                     int             `yaml:"port"`
+		ExternalHostName         string          `yaml:"externalHostName"`
+		ExternalPort             int             `yaml:"externalPort"`
+		SecureIO                 bool            `yaml:"secureIO"`
+		Gossip                   bool            `yaml:"gossip"`
+		ConnectTimeout           time.Duration   `yaml:"connectTimeout"`
+		MasterKey                string          `yaml:"masterKey"`
+		Relay                    string          `yaml:"relay"` // could be `active`, `nat`, `disable`
+		ConnLowWater             int             `yaml:"connLowWater"`
+		ConnHighWater            int             `yaml:"connHighWater"`
+		RateLimiterLRUSize       int             `yaml:"rateLimiterLRUSize"`
+		BlackListLRUSize         int             `yaml:"blackListLRUSize"`
+		BlackListErrorThreshold  int             `yaml:"blackListErrorThreshold"`
+		BlackListCleanupInterval time.Duration   `yaml:"blackListCleanupInterval"`
+		ConnGracePeriod          time.Duration   `yaml:"connGracePeriod"`
+		EnableRateLimit          bool            `yaml:"enableRateLimit"`
+		RateLimit                RateLimitConfig `yaml:"rateLimit"`
+		PrivateNetworkPSK        string          `yaml:"privateNetworkPSK"`
+	}
+
+	// RateLimitConfig all numbers are per second value.
+	RateLimitConfig struct {
+		GlobalUnicastAvg   int `yaml:"globalUnicastAvg"`
+		GlobalUnicastBurst int `yaml:"globalUnicastBurst"`
+		PeerAvg            int `yaml:"peerAvg"`
+		PeerBurst          int `yaml:"peerBurst"`
+	}
+)
 
-// DefaultRatelimitConfig is the default rate limit config
-var DefaultRatelimitConfig = RateLimitConfig{
-	GlobalUnicastAvg:   300,
-	GlobalUnicastBurst: 500,
-	PeerAvg:            300,
-	PeerBurst:          500,
-}
+var (
+	// DefaultConfig is a set of default configs
+	DefaultConfig = Config{
+		HostName:                 "127.0.0.1",
+		Port:                     30001,
+		ExternalHostName:         "",
+		ExternalPort:             30001,
+		SecureIO:                 false,
+		Gossip:                   false,
+		ConnectTimeout:           time.Minute,
+		MasterKey:                "",
+		Relay:                    "disable",
+		ConnLowWater:             200,
+		ConnHighWater:            500,
+		RateLimiterLRUSize:       1000,
+		BlackListLRUSize:         1000,
+		BlackListErrorThreshold:  3,
+		BlackListCleanupInterval: 600 * time.Second,
```

Comment:
unify naming to either "blocklist" or "blacklist"

Reply:
updated